### ISSUE TITLE
[Feature] Add service/button to perform router's internal speed test and store results

### ISSUE THREAD

**Issue by Valdorama** [Vaskivskyi/ha-asusrouter#704]
### The problem

The official HA SpeedTest integration is fairly unreliable (and I believe it uses the wrong API?). Asus routers have a built-in internet speed test feature:

![image](https://github.com/Vaskivskyi/ha-asusrouter/assets/34754485/692f8751-b3cf-49a4-a0be-578e12c20640)

It would be great if this integration could provide an ability to trigger the router's speed test and store the results. It could be provided as a service or a button, with the most-recent result stored in attributes (download, upload, ping).

Users could then create automations to periodically run the speed test and track performance over time.

### Specific device(s)?

None

### Firmware type

None

**Comment by Vaskivskyi** [comment]
Hello. Yes, this should be possible. I will check it

**Comment by Vaskivskyi** [comment]
Yes, the feature can be implemented. It might even come in the next version since it does not seem to be complicated

**Comment by Valdorama** [comment]
Thanks, that would be much appreciated.

**Comment by rac146** [comment]
Hello! Is there any way to trigger a new speed test via this sensor/service?

**Comment by Vaskivskyi** [comment]
Unfortunately, up till now, I didn't manage to make the testing work. Speedtest does not like any "automated" triggering. That's why sensors are not added to the stable integration branch (release) - there is no point yet without the possibility to run the test.

I will post updates as soon as there is any progress.

**Comment by Vaskivskyi** [comment]
The feature is finally coming to the backend library with v2. The merge of v2 into the integration can be tracked in #1297. Please watch that thread. When the migration is done, new features will be gradually implemented into the library
